### PR TITLE
MR-3414: Test rates refactor work

### PR DESCRIPTION
**Pull request by JasonLin0991**
## Summary
[MR-3414](https://qmacbis.atlassian.net/browse/MR-3414)

Unlock tests for rates refactor work.

### CODE REVIEW

File: services/app-api/src/postgres/contractAndRates/unlockContract.test.ts [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        const draftContract = must(await findDraftContract(client, contract.id))
+
+        if (draftContract === undefined) {
+            if (draftContract === undefined) {
+                throw Error('Contract data was undefined')
+            }
+        }
+
+        // Rate revision should be connected to contract
+        expect(draftContract.rateRevisions[0].id).toEqual(submittedRateRev.id)
```
Comment (macrael) [inline services/app-api/src/postgres/contractAndRates/unlockContract.test.ts:71]: I _think_ this shouldn't work? draftContract is only going to have draftRates. Might not be returning the right thing, here.

**Review by macrael — APPROVED** [review]
these tests are excellent

File: services/app-api/src/postgres/contractAndRates/unlockContract.test.ts (hunk 1 above)
Comment (JasonLin0991) [reply services/app-api/src/postgres/contractAndRates/unlockContract.test.ts:71]: It looks like its working?

Screenshot 2023-06-28 at 11 52 21 AM

Submitted Rate
```json
{
   "id":"ca6e933f-623c-4da0-b5df-3879be6ae6c1",
   "revisions":[
      {
         "id":"25cc195b-d389-4931-a07a-f68de2974eb0",
         "submitInfo":{
            "updatedAt":"2023-06-28T15:51:16.018Z",
            "updatedBy":"4b5a3ab3-ae41-4d8b-b0b1-910925bc1b03",
            "updatedReason":"Rate A 1.0 submit"
         },
         "revisionFormData":"Rate 1.0",
         "contractRevisions":[
            
         ]
      }
   ]
}
```

Draft Contract
```json
{
   "id":"6742d8b4-d3a1-4874-a431-9ebb76557c48",
   "contractFormData":"Connecting rate",
   "rateRevisions":[
      {
         "id":"25cc195b-d389-4931-a07a-f68de2974eb0",
         "revisionFormData":"Rate 1.0"
      }
   ]
}
```

Comment (macrael) [reply services/app-api/src/postgres/contractAndRates/unlockContract.test.ts]: Interesting, I'll take a look at this myself after it merges and think about it

**Review by macrael — APPROVED** [review]

**Pull request merged**